ResearchKit/Common/ORKNavigableOrderedTask.h

```diff
@@ -99,7 +106,46 @@ ORK_CLASS_AVAILABLE
 @property (nonatomic, copy, readonly) NSDictionary<NSString *, ORKStepNavigationRule *> *stepNavigationRules;
 
 /**
- Determines whether the task should report its progress as a linear ordered task or not. 
+ Adds a skip step navigation rule for a step identifier.
+ 
+ The rule will be used to decide if the identified step needs to be skipped. You cannot add two
+ different skip navigation rules to the same step identifier; only the most recently added rule is
+ kept.
+ 
+ @param skipStepNavigationRule      The skip step navigation rule to be used to determine if the
+                                        step should be skipped. A strong reference to the rule is
+                                        kept by the task.
+ @param stepIdentifier              The identifier of the step that is checked against the skip
+                                        rule.
+ */
+- (void)setSkipNavigationRule:(ORKSkipStepNavigationRule *)skipStepNavigationRule forStepIdentifier:(NSString *)stepIdentifier;
+
+/**
+ Returns the skip step navigation rule associated with a step identifier,  or `nil` if there is no
+ skip rule associated with that step identifier.
+ 
+ @param stepIdentifier      The identifier of the step whose skip rule you want to retrieve.
+ 
+ @return A skip step navigation rule, or `nil` if the step identifier has none.
+ */
+- (ORKSkipStepNavigationRule *)skipNavigationRuleForStepIdentifier:(NSString *)stepIdentifier;
+
+/**
+ Removes the skip step navigation rule, if any, associated with the specified step identifier.
+ 
+ @param stepIdentifier   The identifier of the step whose rule is to be removed.
+ */
+- (void)removeSkipNavigationRuleForStepIdentifier:(NSString *)stepIdentifier;
+
+/**
+ A dictionary of step navigation rules in the task, keyed by trigger step identifier.
+ 
+ Each object in the dictionary should be a `ORKStepNavigationRule` subclass.
+ */
+@property (nonatomic, copy, readonly) NSDictionary<NSString *, ORKSkipStepNavigationRule *> *skipStepNavigationRules;
```

Comment:
Is this attribute nullable? If it is not nullable, might need to write a getter in the implementation to create an empty dictionary in case  _ skipStepNavigationRules is still nil.